src/libm/sleefdp.c

```diff
@@ -1135,14 +1154,39 @@ static INLINE CONST Sleef_double2 logk(double d) {
 }
 
 EXPORT CONST double xlog_u1(double d) {
-  Sleef_double2 s = logk(d);
-  double x = s.x + s.y;
+  Sleef_double2 x;
+  double m, t, x2;
+  int e;
 
-  if (xisinf(d)) x = INFINITY;
-  if (d < 0) x = NAN;
-  if (d == 0) x = -INFINITY;
+  int o = d < DBL_MIN;
+  if (o) d *= (double)(1LL << 32) * (double)(1LL << 32);
+      
+  e = ilogb2k(d * (1.0/0.75));
+  m = ldexp3k(d, -e);
 
-  return x;
+  if (o) e -= 64;
+  
+  x = dddiv_d2_d2_d2(ddadd2_d2_d_d(-1, m), ddadd2_d2_d_d(1, m));
+  x2 = x.x * x.x;
+
+  t = 0.1532076988502701353e+0;
+  t = mla(t, x2, 0.1525629051003428716e+0);
+  t = mla(t, x2, 0.1818605932937785996e+0);
+  t = mla(t, x2, 0.2222214519839380009e+0);
+  t = mla(t, x2, 0.2857142932794299317e+0);
+  t = mla(t, x2, 0.3999999999635251990e+0);
+  t = mla(t, x2, 0.6666666666667333541e+0);
+  
+  Sleef_double2 s = ddadd2_d2_d2_d2(ddmul_d2_d2_d(dd(0.693147180559945286226764, 2.319046813846299558417771e-17), (double)e),
```

Comment:
This is the problem discussed in the following page.

http://stackoverflow.com/questions/3995940/casting-one-c-structure-into-another

We cannot expose double2 data type in SLEEF API since it is too common.

Using typedef like below did not work.

typedef double2 Sleef_double2;

Do you have any idea to improve this?